### PR TITLE
Deprecate roi objects

### PR DESCRIPTION
**Pull request by HaleySchuhl**
**Describe your changes**
Deprecate `pcv.roi_objects` but also update `pcv.record_size_marker` to remove the code which using `object_composition`. 

**Type of update**
Is this a:
* Deprecation updated to release-4.0 branch 

**Associated issues**
#945 

**Additional context**
Merge #1229 first. This 
**For the reviewer**
See [this page](https://plantcv.readthedocs.io/en/latest/pr_review_process/) for instructions on how to review the pull request.
- [ ] PR functionality reviewed in a Jupyter Notebook
- [ ] All tests pass
- [ ] Test coverage remains 100%
- [ ] Documentation tested
- [ ] New documentation pages added to `plantcv/mkdocs.yml`
- [ ] Changes to function input/output signatures added to `updating.md`
- [ ] Code reviewed
- [ ] PR approved

### CODE REVIEW

**Comment by nfahlgren** [comment]
I ran `mkdocs serve --theme=readthedocs` to build the documentation locally and it reports that there are doc pages with references to `roi_objects.md`:

`analysis_approach.md`
`params.md`
`roi_custom.md`
`tutorials/hyperspectral_tutorial.md`

File: plantcv/plantcv/report_size_marker_area.py [hunk 1]
```diff
@@ -90,6 +84,7 @@ def report_size_marker_area(img, roi, marker='define', objcolor='dark', thresh_c
             fatal_error('thresh_channel and thresh must be defined in detect mode')
     elif marker.upper() == "DEFINE":
         # Identify contours in the masked image
+        # Are we using a mask to make a new mask here?? lol
```
Comment (nfahlgren) [inline plantcv/plantcv/report_size_marker_area.py:87]: Looks like yes. I think in this section we could just do `marker_mask = roi_mask` and remove the rest

**Review by nfahlgren — CHANGES_REQUESTED** [review]
In addition to the code comments, please use the flake8 linter suggestions in VSCode to fix some of the formatting/style issues

**Comment by HaleySchuhl** [comment]
> I ran `mkdocs serve --theme=readthedocs` to build the documentation locally and it reports that there are doc pages with references to `roi_objects.md`:
> 
> `analysis_approach.md` `params.md` `roi_custom.md` `tutorials/hyperspectral_tutorial.md`

I've addressed everything aside from the tutorial!

**Pull request merged**